### ISSUE TITLE
Failed to replicate the experiment on the coco dataset using yolo11

### ISSUE THREAD

**Issue by Acui001** [ultralytics/ultralytics#17048]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

train code:
from ultralytics import YOLO
model = YOLO("yolo11n.pt")  # load a pretrained model (recommended for training)
results = model.train(data="coco.yaml", epochs=100, imgsz=640, device=[0])

I didn't change any parameters of cfg/default.yaml，Then the fine-tuning experiment was carried out using yolo11n.pt
The final parameter mAP50-95 is only 37.7, which is still 1.8 different from what you published
So I need to use the following code to find the best parameters and then fine-tune it with best_hyperparameters.yaml?
from ultralytics import YOLO
model = YOLO("yolo11n.pt")
model.tune(data="coco.yaml", epochs=30, iterations=300, optimizer="AdamW", plots=False, save=False, val=False)

Or do I need to start from scratch to train for replicate the experiment? Is it because fine-tuning cannot reproduce the original accuracy?

Hope to get your reply

**Comment by Y-T-G** [comment]
For replication, you usually start from scratch.

And follow this:
https://github.com/ultralytics/ultralytics/issues/16607#issuecomment-2388336537

If you want to fine-tune, you can follow this:
https://www.reddit.com/r/Ultralytics/comments/1eolwl8/the_correct_way_to_train_from_a_previously/

**Comment by Acui001** [comment]
> For replication, you usually start from scratch.
> 
> And follow this: ： [#16607 (comment)](https://github.com/ultralytics/ultralytics/issues/16607#issuecomment-2388336537)
> 
> If you want to fine-tune, you can follow this:： https://www.reddit.com/r/Ultralytics/comments/1eolwl8/the_correct_way_to_train_from_a_previously/

you mean I should start from scratch and And follow this:

![tain_tip](https://github.com/user-attachments/assets/a16485a5-7020-4094-9d89-bf877541b135)
Another thing is that if I see that the cfg/default.yaml parameter is epoch=100,batch=16, and the website:

> https://docs.ultralytics.com/yolov5/tutorials/tips_for_best_training_results/#model-selection

 says that epoch training should be tried from 300 to 600, do I still need to try this if I want to repeat it or do I just default to default.yaml? I just want to know if the parameters can be completely reproduced after changing according to the table.

**Comment by Y-T-G** [comment]
You can find the args used during training from the model:

```python
from ultralytics import YOLO, ASSETS

model = YOLO("yolo11n.pt")
print(model.ckpt["train_args"]
``` 

You can follow the same args for training from scratch but for augmentations follow this:
https://github.com/ultralytics/ultralytics/issues/16607#issuecomment-2388336537

**Comment by Acui001** [comment]
> You can find the args used during training from the model:
> 
> ```python
> from ultralytics import YOLO, ASSETS
> 
> model = YOLO("yolo11n.pt")
> print(model.ckpt["train_args"])
> ```
> 
> You can follow the same args for training from scratch but for augmentations follow this: [#16607 (comment)](https://github.com/ultralytics/ultralytics/issues/16607#issuecomment-2388336537)

I created such a parameter file according to your guidance (see the attachment args.yaml) and used the augmentation https://github.com/ultralytics/ultralytics/issues/16607#issuecomment-2388336537 you mentioned, but the experimental results (see the attachments result.csv and result.png) are still 0.5% different from the 39.5 you published. Is this a normal error or is there any parameter I am still wrong? Hope to get your reply.

[Uploading train68.zip…]()

**Comment by Y-T-G** [comment]
Your batch size is different. You used two GPUs. The batch size would become 128/2 = 64 for each GPU. But the original uses a single GPU, so it would be 128. If you want 128 with a single GPU, you would have to set it to 256. So 256/2 = 128 each GPU. But it might still be different because it's not exactly equivalent to single GPU training.

**Comment by sunsmarterjie** [comment]
Hi bro, did you reproduce yolov11n's results? I also can not reproduce it with the correct batch size: https://github.com/ultralytics/ultralytics/issues/17165

Anyone can help to find the reason?

**Comment by Acui001** [comment]
> Hi bro, did you reproduce yolov11n's results? I even can not reproduce it with the correct batch size: #17165
> 
> Anyone can help to find the reason?

Officials say training with a single GPU with a batch of 128,but my team's GPU(4090) not satisfy this.So I give up to replicate the experiment

**Comment by sunsmarterjie** [comment]
Based on my extensive experience with experiments on various versions of YOLO, a batch size of 64 or 128 should not cause a discrepancy as large as 0.5-0.7. I recommend that the official team strictly use the currently open-sourced code to reproduce the reported results. I believe this will help identify the reasons for the reproduction issue.

**Comment by glenn-jocher** [comment]
Thank you for your feedback. Reproducing our results depends on strict adherence to the specific training parameters, including hardware configuration, batch size, and augmentation settings, as detailed in our documentation. Minor differences in hardware or settings often lead to slight discrepancies. Please ensure absolute alignment with the published configuration. If the issue persists, feel free to share a Minimum Reproducible Example (MRE) for further review.

**Comment by Acui001** [comment]
> Based on my extensive experience with experiments on various versions of YOLO, a batch size of 64 or 128 should not cause a discrepancy as large as 0.5-0.7. I recommend that the official team strictly use the currently open-sourced code to reproduce the reported results. I believe this will help identify the reasons for the reproduction issue.

https://github.com/ultralytics/ultralytics/issues/22092#issue-3419281908
Hi bro, when you were reproducing the experiment, did you encounter such a huge accuracy gap when running on multiple GPUs? I'm currently out of ideas.

**Comment by glenn-jocher** [comment]
Thanks for the follow‑up—multi‑GPU can introduce small convergence differences; to isolate the gap please run a single‑GPU, deterministic baseline and open a new issue (this thread is closed/stale) with full diagnostics for visibility.

```python
# Deterministic single-GPU baseline
import torch
from ultralytics import YOLO

torch.backends.cudnn.benchmark = False
torch.backends.cudnn.deterministic = True
torch.use_deterministic_algorithms(True)

model = YOLO("yolo11n.yaml")  # train from scratch to replicate published results
model.train(data="coco.yaml", imgsz=640, epochs=600, batch=128, device=0, workers=0, amp=False, seed=0)
```

Please include: the text output of `yolo checks`, `from ultralytics.utils.checks import collect_system_info; print(collect_system_info())`, your exact global batch (per‑GPU × GPUs), `results.csv` and `args.yaml`, and the output of `model.ckpt['train_args']`. For reference, see the Train mode guide at [Model Training with Ultralytics YOLO](https://docs.ultralytics.com/modes/train/) and the GPU/setup tips in [Troubleshooting Common YOLO Issues](https://docs.ultralytics.com/guides/yolo-common-issues/).
